### PR TITLE
MNTOR-2683: cascade del on "email_addresses" table's fk

### PR DESCRIPTION
**Pull request by mansaj**
# References: 
Jira: MNTOR-2683

# Description
We are seeing some sql errors on del with foreign key constraints. This PR sets cascade del on `subscriber_id` FK in the table `email_addresses` to ensure that when a subscriber is deleted, the corresponding additional emails will also

# How to test
* Delete a subscriber
* Not seeing fk errors in sql

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/db/tables/subscribers.js [hunk 1]
```diff
@@ -337,38 +337,6 @@ async function setWaitlistsJoined (options) {
 }
 /* c8 ignore stop */
 
-/**
- * @param {import('../../app/(nextjs_migration)/(authenticated)/user/breaches/breaches.js').Subscriber} subscriber
- */
-// Not covered by tests; mostly side-effects. See test-coverage.md#mock-heavy
-/* c8 ignore start */
-async function removeSubscriber (subscriber) {
```
Comment (mansaj) [inline src/db/tables/subscribers.js:345]: Unrelated, but seems like unused code? Doing some tech debt cleanup

Comment (rhelmer) [reply src/db/tables/subscribers.js:345]: Hm, is it a duplicate of `deleteSubscriber`?

Comment (mansaj) [reply src/db/tables/subscribers.js:345]: I think so, it might be older. I searched for it in the codebase and there's no other file referencing it

Comment (mansaj) [reply src/db/tables/subscribers.js:345]: I'd like to get this into heroku to test it via fxa

**Review by rhelmer — APPROVED** [review]

**Pull request merged**